core/market/decentralized/src/testing/mock_node.rs

```diff
@@ -94,9 +94,20 @@ impl MarketsNetwork {
     /// Remember that test_name should be unique between all tests.
     /// It will be used to create directories and GSB binding points,
     /// to avoid potential name clashes.
-    pub async fn new(test_name: &str) -> Self {
+    pub async fn new(test_name: Option<&str>) -> Self {
         let _ = env_logger::builder().try_init();
-        let test_dir = prepare_test_dir(&test_name).unwrap();
+        // level 1 is this function.
+        // level 2 is <core::future::from_generator::GenFuture<T> as
+        // core::future::future::Future>::poll::XXX> (async)
+        // We want to know the caller.
+        let mut bn = crate::testing::generate_backtraced_name(Some(3));
+        // Special case for mac&windows
+        if !bn.starts_with("test_") {
```

Comment:
hackish. Can you extend comment and describe why do we expect `test_` here? For future readers -- also ourselves ;)